bot.py

```diff
@@ -37,43 +42,39 @@ async def on_ready(self):
         # tree.copy_global_to(guild=guild)
         # await tree.sync(guild=guild)
 
-    async def on_button_click(self, interaction):
-        logging.info(f"button clicked: {interaction}")
-
     async def on_message(self, message: Message):
         # Don't respond to ourselves
         if message.author == self.user:
             return
-        # Only look in the active stream channel
-        channels_to_listen_to = {STREAM_CHAT_ID, WELCOME_CHAT_ID}
-        if message.channel.id not in channels_to_listen_to:
-            return
 
-        if message.channel.id == STREAM_CHAT_ID:
-            DB().accrue_channel_points(message.author.id, message.author.roles)
+        # Server Subscription message
+        if message.channel.id == WELCOME_CHAT_ID and message.type.value == 25:
+            # fetch extra attached message info
+            raw_msg = await self.http.get_message(channel_id=message.channel.id, message_id=message.id)
+
+            role_sub_data = raw_msg.get("role_subscription_data")
+            if role_sub_data is None:
+                return logging.error(f"Unable to get role subscription data for message: {message.id}")
 
-        if message.channel.id == WELCOME_CHAT_ID:
-            premium_ids = map(
-                int,
-                [
-                    Config.CONFIG["Discord"]["Tier1RoleID"],
-                    Config.CONFIG["Discord"]["Tier2RoleID"],
-                    Config.CONFIG["Discord"]["Tier3RoleID"],
-                ],
-            )
-
-            role_name = None
-            for role_id in premium_ids:
-                role = discord.utils.get(message.author.roles, id=role_id)
-                if role is not None:
-                    role_name = role.name
-                    break
-
-            if role_name is not None:
-                await self.get_channel(STREAM_CHAT_ID).send(
-                    f"Thank you {message.author.mention} for joining {role_name}!"
-                )
+            # comes in like this -> 'tier_name': 'THE ONES WHO KNOW membership',
+            # trim " membership" and get the actual role
+            role_name = role_sub_data.get("tier_name", "").rstrip(" membership")
+            role = next(filter(lambda x: x.name.startswith(role_name) and x.id in PREMIUM_IDS, message.guild.roles), None)
```

Comment:
This looks like you're trying to look up which role out of `PREMIUM_IDS` the member now has. Is there a reason we can't create a map of role name to ID?
```python
PREMIUM_IDS = {
    "THE ONES WHO KNOW": Config.CONFIG["Discord"]["Tier1RoleID"],
    ...
}
```

Reply:
honestly just checking to see whether they are in the premium roles. I debated removing it entirely, wdyt

Reply:
We wouldn't get this message type if it's not a premium role, right?

Reply:
yeah, this is really to mitigate a scenario where there's another role starting with `THE ONES WHO KNOW` etc

Reply:
Just tell Christian not to make that role 🧠